api/command.go

```diff
@@ -17,630 +16,386 @@ import (
-		command.AddSuggestion(&model.SuggestCommand{Suggestion: cmd2, Description: c.T("api.command.load_test_posts_command.fuzz.description")})
-	} else if strings.Index(cmd2, command.Command) == 0 {
-		command.AddSuggestion(&model.SuggestCommand{Suggestion: cmd2, Description: c.T("api.command.load_test_posts_command.fuzz.description")})
+	if result := <-Srv.Store.Command().Update(cmd); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
+		w.Write([]byte(result.Data.(*model.Command).ToJson()))
 	}
-
-	return false
 }
 
-func loadTestUrlCommand(c *Context, command *model.Command) bool {
-	cmd := cmds["loadTestCommand"] + " url"
-
-	if strings.Index(command.Command, cmd) == 0 && !command.Suggest {
-		url := ""
+func deleteCommand(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !*utils.Cfg.ServiceSettings.EnableCommands {
+		c.Err = model.NewLocAppError("deleteCommand", "api.command.disabled.app_error", nil, "")
+		c.Err.StatusCode = http.StatusNotImplemented
+		return
+	}
 
-		parameters := strings.SplitN(command.Command, " ", 3)
-		if len(parameters) != 3 {
-			c.Err = model.NewLocAppError("loadTestUrlCommand", "api.command.load_test_url_command.url.app_error", nil, "")
-			return true
-		} else {
-			url = parameters[2]
+	if *utils.Cfg.ServiceSettings.EnableOnlyAdminIntegrations {
+		if !(c.IsSystemAdmin() || c.IsTeamAdmin()) {
+			c.Err = model.NewLocAppError("deleteCommand", "api.command.admin_only.app_error", nil, "")
+			c.Err.StatusCode = http.StatusForbidden
+			return
 		}
+	}
 
-		// provide a shortcut to easily access tests stored in doc/developer/tests
-		if !strings.HasPrefix(url, "http") {
-			url = "https://raw.githubusercontent.com/mattermost/platform/master/doc/developer/tests/" + url
+	c.LogAudit("attempt")
 
-			if path.Ext(url) == "" {
-				url += ".md"
-			}
-		}
-
-		var contents io.ReadCloser
-		if r, err := http.Get(url); err != nil {
-			c.Err = model.NewLocAppError("loadTestUrlCommand", "api.command.load_test_url_command.file.app_error", nil, err.Error())
-			return false
-		} else if r.StatusCode > 400 {
-			c.Err = model.NewLocAppError("loadTestUrlCommand", "api.command.load_test_url_command.file.app_error", nil, r.Status)
-			return false
-		} else {
-			contents = r.Body
-		}
+	props := model.MapFromJson(r.Body)
 
-		bytes := make([]byte, 4000)
+	id := props["id"]
+	if len(id) == 0 {
+		c.SetInvalidParam("deleteCommand", "id")
+		return
+	}
 
-		// break contents into 4000 byte posts
-		for {
-			length, err := contents.Read(bytes)
-			if err != nil && err != io.EOF {
-				c.Err = model.NewLocAppError("loadTestUrlCommand", "api.command.load_test_url_command.reading.app_error", nil, err.Error())
-				return false
-			}
+	if result := <-Srv.Store.Command().Get(id); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
+		if c.Session.TeamId != result.Data.(*model.Command).TeamId && c.Session.UserId != result.Data.(*model.Command).CreatorId && !c.IsTeamAdmin() {
```

Comment:
This looks weird to me. Are team members supposed to be able to delete other member's commands? This also looks like team admins are able to delete commands made by other teams